apps/docs/docs/core/core.md

```diff
@@ -95,3 +95,26 @@ export const contract = c.router({
   },
 });
 ```
+
+## Response Types
+
+If you need to quickly extract the Response types from a given Contract, you can use `getRouteResponses`.
+
+:warning: This does not support nested contracts yet. You need to pass in a defined contract. 
+
+`getRouteResponses(router)` returns an empty object with defined response types for a contract's HTTP method. 
+
+```typescript
+import { getRouteResponses } from '@ts-rest/core'
+import { contract } from './contract'
+
+//  initContract() must called prior
+const contractResponses = getRouteResponses(contract)
+type ResponseShapes = typeof contractResponses
```

Comment:
Quick question, why is this feature implemented as a function rather than just a generic type?

I saw you already made `ApiResponseForRoute`, we could just make a mapped type right? e.g. this could be a resulting type

```typescript

type Test = ResponseShapes<typeof MyAppRouter>
//       ^?
// {
//  "getPosts" : ApiResponseForRoute<...>
// }
```

Reply:
Because you call the function, which just returns {}, then you only use it at compile time with the typeof, feels a bit superfluous maybe?

Reply:
Or even a generic that could be used like

```
async getAThing(): RouteResponse<MyApi["getAThing"]> 
```

Or something where you could pass in any `AppRoute`, not 100% sure how I prefer this, lots of good options

Reply:
So I implemented a runtime function to allow the user to use this on any runtime `router` already defined. For instance, if you initialize `router()` already and want to extract the response types somewhere else in your code, you could use this generic function to pull those types out at any time. The function is also used in the `ts-rest-nest.ts` implementation here: 
```
  return {
    controllerShape: {} as NestControllerShapeFromAppRouter<T>,
    routeShapes: {} as NestAppRouteShape<T>,
    responseShapes: getRouteResponses(router), <---
    route: router,
  };
```
I agree we could use a type, but I figured it's a better DX to do something like
`getRouteResponses(router)` versus `ApiResponseForRoute<type of router>` in your code. 

Regarding the generic type, I was limiting it to just response types to limit scope creep/major changes to the core.